src/action.c

```diff
@@ -144,11 +169,31 @@ action_str_from_arg(struct action_arg *arg)
 	return ((struct action_arg_str *)arg)->value;
 }
 
+static char *
+str_from_key(struct action *action, const char *key, char *default_value)
+{
+	assert(key);
+	struct action_arg *arg;
+	wl_list_for_each(arg, &action->args, link) {
+		if (!strcmp(key, arg->key)) {
+			return action_str_from_arg(arg);
+		}
+	}
+	return default_value;
+}
+
 static bool
-action_bool_from_arg(struct action_arg *arg)
+bool_from_key(struct action *action, const char *key, bool default_value)
```

Comment:
I am not sure about the naming here, I kind of miss the `arg` part in `bool_from_key` / `str_from_key`.
Suggestion: `get_arg_value_bool()` / `get_arg_value_str()`. Open to different naming though.

Reply:
Renamed to get_arg_value_{bool,str}.
Agree, it seems better